naga/src/front/wgsl/lower/mod.rs

```diff
@@ -2208,6 +2210,15 @@ impl<'source, 'temp> Lowerer<'source, 'temp> {
                     .collect::<Result<Vec<_>, _>>()?;
 
                 let has_result = ctx.module.functions[function].result.is_some();
+                let must_use = ctx.module.functions[function]
+                    .result
+                    .as_ref()
+                    .is_some_and(|result| result.must_use);
```

Comment:
Since all the validation happens in the frontend it doesn't seem necessary to have `FunctionResult.must_use` as part of the IR.

Would it work to add the `must_use` bool to `LoweredGlobalDecl::Function` instead?

Reply:
Sounds good. I've made that change, and reverted the other related changes, ie the helper functions FunctionResult::new() and FunctionResult::bound(), and the IR snapshots